src/valkeymodule.h

```diff
@@ -831,12 +831,46 @@ typedef struct ValkeyModuleScriptingEngineMemoryInfo {
     size_t engine_memory_overhead;
 } ValkeyModuleScriptingEngineMemoryInfo;
 
-/* The callback function called when `FUNCTION LOAD` command is called to load
- * a library of functions.
- * This callback function evaluates the source code passed to `FUNCTION LOAD`
- * and registers the functions declared in the source code.
+typedef enum ValkeyModuleScriptingEngineSubsystemType {
+    VMSE_EVAL,
+    VMSE_FUNCTION,
+    VMSE_ALL
+} ValkeyModuleScriptingEngineSubsystemType;
+
+typedef enum ValkeyModuleScriptingEngineExecutionState {
+    VMSE_STATE_EXECUTING,
+    VMSE_STATE_KILLED,
+} ValkeyModuleScriptingEngineExecutionState;
```

Comment:
Can there be any other states in the future? PAUSED?

If it's just about whether it's killed or not, we can consider `bool`. WDYT?

Reply:
Keep the enum. I agree, enums are more informative.